src/kademlia/dht_storage.cpp

```diff
@@ -453,6 +469,19 @@ namespace
 			touch_item(i->second, addr);
 		}
 
+		int get_infohashes_sample(entry& item) const override
+		{
+			item["interval"] = clamp(m_settings.sample_infohashes_interval
+				, 0, sample_infohashes_interval_max);
+			item["num"] = int(m_map.size());
+
+			std::vector<sha1_hash> const& samples = m_infohashes_sample.samples;
```

Comment:
I think it would be slightly nicer if the samples were computed lazily. i.e. right before this line, check if the samples have expired or not. if they have, call ``refresh_infohashes_sample()``.

Or call that function unconditionally here, and it checks to see if the sample is still "fresh".